internal/update/update.go

```diff
@@ -0,0 +1,120 @@
+// From https://github.com/cli/cli/blob/trunk/internal/update/update.go
+package update
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"gopkg.in/yaml.v3"
+)
+
+const repo = "algolia/cli"
+
+var gitDescribeSuffixRE = regexp.MustCompile(`\d+-\d+-g[a-f0-9]{8}$`)
+
+// ReleaseInfo stores information about a release
+type ReleaseInfo struct {
+	Version     string    `json:"tag_name"`
+	URL         string    `json:"html_url"`
+	PublishedAt time.Time `json:"published_at"`
+}
+
+type StateEntry struct {
+	CheckedForUpdateAt time.Time   `yaml:"checked_for_update_at"`
+	LatestRelease      ReleaseInfo `yaml:"latest_release"`
+}
+
+// CheckForUpdate checks whether this software has had a newer release on GitHub
+func CheckForUpdate(client *http.Client, stateFilePath, currentVersion string) (*ReleaseInfo, error) {
+	stateEntry, _ := getStateEntry(stateFilePath)
+	if stateEntry != nil && time.Since(stateEntry.CheckedForUpdateAt).Hours() < 24 {
```

Comment:
Remember that the users would be notified only once per day anyway. I agree that we are pushing new version too frequently, but we are in beta, it will be slower avec the release.